### PR TITLE
Update binaries page after user testing

### PR DESCRIPTION
**Pull request by nrichers**
This PR makes some updates to the instructions for [getting started with the binaries](https://questdb.io/docs/get-started/binaries/) after user testing.

## Overview of the changes

- **General edits**. Tweaks to use present tense, lead with the specific OS in the opening sentences of the intro, apply some writing minimalism, and clarify some headings.
- **Separate the Java prerequisite from downloading the QuestDB binaries**. Currently, downloading the binaries and the Java 11 prerequisite are interspersed at the same heading level, but the real-world sequence is likely checking the Java version > downloading tarballs > extracting > running QuestDB. The new structure reflects that sequence. 
- **Clarify tab labels for extracting tarballs**. Current labels don't mention that some labels apply to extracting the QuestDB tarballs and so it _almost_ looks like some instructions are missing. They aren't missing, it's just the tab labels.
- **Separate out upgrading from running QuestDB**. Currently, the steps to actually run QuestDB are shown only under upgrading (or you have to refer to the command line reference). For the sake of making sure that you end the page with an up and running QuestDB instance, I added a section.
- **Tweak the upgrading commands shown**. As-is, they omit the middle step that includes the actual upgrading. This change is a _hack_, and ideally you would have numbered steps with separate command snippets. It's an interim fix — it might make sense to move upgrading into a separate section in the future as it's not essential to getting started. 

## Output

Here's what `yarn start` shows on localhost:

image
image

### CODE REVIEW

**Comment by nrichers** [comment]
Looks like the CI checks are failing with `Error: Timeout reached: Unable to connect to https://deploy-preview-1193--unruffled-blackwell-31bfb2.netlify.app`. The changes build fine locally, so I am not sure what the issue with the preview is.

**Review by amyshwang — COMMENTED** [review]
Hi! Thank you for the PR 👋 
I have not gone through this page thoroughly before and you're absolutely right that the section on "upgrade" is an interim solution.

I made some suggestion and I think CI is also working now.

**Comment by insmac** [comment]
> Looks like the CI checks are failing with `Error: Timeout reached: Unable to connect to https://deploy-preview-1193--unruffled-blackwell-31bfb2.netlify.app`. The changes build fine locally, so I am not sure what the issue with the preview is.
> 
> EDIT: I hadn't pushed a commit to fix a code snippet from my local branch, maybe that is the issue. EDIT EDIT: Still fails with the same preview error.

@nrichers this was due to the PR build needed to be approved first within our Netlify dashboard - it should be all good now, and tests are passing.

File: docs/get-started/binaries.md [hunk 1]
```diff
@@ -9,63 +9,54 @@ description:
...
-This page describes how to install and use QuestDB via binaries. QuestDB comes
-with a script `questdb.sh` for Linux/FreeBSD and an executable `questdb.exe` for
-Windows. If you are looking for macOS, please check our
-[Homebrew](/docs/get-started/homebrew) section.
-
-## Download
-
-You can find the latest binaries on the [Get QuestDB](/get-questdb) page. The
-release notes are on our [GitHub release]({@githubUrl@}/releases) page.
+This page describes how to install and run QuestDB via binaries. On Linux or FeeBSD, you run QuestDB with the `questdb.sh` script. On Windows, you run the `questdb.exe` executable. For macOS, check out [Homebrew](/docs/get-started/homebrew).
```
[the remark this answers is not in the document]
Comment (nrichers) [reply docs/get-started/binaries.md:12]: Is this a style guide preference for you? Consistency generally matters, but editors I worked with have recommend that you address the reader directly if it's an action they have to perform (such as running a binary). The text can totally be changed to something else.

File: docs/get-started/binaries.md [hunk 2]
```diff
@@ -214,24 +202,56 @@ runtime is packaged directly with QuestDB.
 
 </Tabs>
 
+## Run QuestDB
```
[the remark this answers is not in the document]
Comment (nrichers) [reply docs/get-started/binaries.md:205]: > Is it a conscious decision to repeat the command here?

The intent is to match what happens in [the parallel Docker topic](https://questdb.io/docs/get-started/docker/), where you are left with a running instance. It's also what the topic title and intro promise to deliver, at least in an 'up and running' sense. You will need this running instance later to perform the steps in "Create my first dataset". 

(Directing people into reference content to perform a necessary task does work, but it breaks one of the core tenets of technical writing, namely strong content typing. It also means you have to follow a link and read through another topic to perform a simple task step when you could just list it here.)

File: docs/get-started/binaries.md [hunk 3]
```diff
@@ -9,63 +9,54 @@ description:
...
+  renderText={(release) => {
+    return (
+      <CodeBlock className="language-shell">
+        {`questdb-${release.name}-no-jre-bin.tar.gz`}
+      </CodeBlock>
+    )
+  }}
+/>
+
+### QuestDB version
```
[the remark this answers is not in the document]
Comment (nrichers) [reply docs/get-started/binaries.md:56]: Both great suggestions, thank you! I was trying to tread at least somewhat lightly and retain some of the original bits, but I agree. Will push a change to include this suggestion.

File: docs/get-started/binaries.md [hunk 4]
```diff
@@ -242,6 +262,9 @@ values={[
 
 ```shell
 ./questdb.sh stop
+
+(Perform the upgrade)
```
[the remark this answers is not in the document]
Comment (nrichers) [reply docs/get-started/binaries.md:266]: Great suggestion, thank you. Will make these changes as suggested.

File: docs/get-started/binaries.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nrichers) [reply docs/get-started/binaries.md]: Pushed 41718a6928c4619d871677b29e9eea262a987bdc to remove the extraneous occurrences of "you".

File: docs/get-started/binaries.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (nrichers) [reply docs/get-started/binaries.md]: Addressed in 452d9625acc04f812bba44ea46bb500ab3f8fdee.

File: docs/get-started/binaries.md [hunk 5]
```diff
@@ -9,63 +9,54 @@ description:
...
 folder.
 
-### Your operating system version
+## Download the binaries
+
+Download the latest binaries from [Get QuestDB](/get-questdb). For release notes, check [GitHub release]({@githubUrl@}/releases).
+
+### Any (no JVM) version
 
 The file is named:
```
Comment (amyshwang) [inline docs/get-started/binaries.md:44]:
```suggestion
For no JVM binaries, the filename is:
```

File: docs/get-started/binaries.md [hunk 6]
```diff
@@ -9,63 +9,54 @@ description:
...
+      <CodeBlock className="language-shell">
+        {`questdb-${release.name}-no-jre-bin.tar.gz`}
+      </CodeBlock>
+    )
+  }}
+/>
+
+### QuestDB version
+
+Depending on your operating system, the file is named:
```
Comment (amyshwang) [inline docs/get-started/binaries.md:58]:
```suggestion
The filename for different operating systems is as the following:
```

**Comment by nrichers** [comment]
@amyshwang thank you for pushing additional commits! Just to keep you posted, I've been experimenting with direct versioned download links for the tarballs. There's an interesting Docusaurus limitation I'm trying to learn more about, details in this comment. I plan to tinker a bit more, if you don't mind keeping this PR open for another few days? (No guarantees I'll sort it out, but it's been an interesting tool chain to tinker with.)

There are a couple of reasons why direct download links are preferable: 

- _Just make it simple for your users._ Direct, versioned download links mean you don't have to go to another page to locate the right files. Currently, you ask users to match file names to what's on the [download page](https://questdb.io/get-questdb,) but that page doesn't actually list any file names unless you hover over tiles. It seems unnecessarily complicated for a simple download.
- _Your FreeBSD instructions are broken._ There currently is no download for that OS listed on https://questdb.io/get-questdb at all, though you do publish a tarball as per the [release page in GitHub](https://github.com/questdb/questdb/releases/tag/6.6.1). It might make sense to fix that download page, but direct download links on this page here would sidestep the issue. 

Onto the Docusaurus limitation ...

Interestingly, interpolating variables in Docusaurus to create versioned download links is actually not as straightforward as it might seem. You can construct a URL with variables no problem, but it is much harder _to use_ that constructed URL inside of `<a href>` rather than a `<codeblock>`. 

For example: 

`https://github.com/questdb/questdb/releases/download/{release.name}/questdb-{release.name}-rt-freebsd-amd64.tar` 

renders just fine as text, e.g.

 `https://github.com/questdb/questdb/releases/download/6.6.1/questdb-6.6.1-rt-freebsd-amd64.tar`

but injecting the variable inside of an HTML link yields a 404 because the URL ends up malformed. 

I found several GitHub issues that corroborated my experience when tinkering with versioned links. As per https://github.com/facebook/docusaurus/issues/395#issuecomment-973675761, you need to use JSX and that's what I've been trying to get to work. It feels like I am close.

**Comment by amyshwang** [comment]
Hi @nrichers sure I will leave this PR for now. 

- It seems to me that we should just add download links to the Get started>binaries page directly - users don't really need to know about the filename...
- Will try sorting out the missing FreeBSD 

If you have time, do keep me posted about Docusaurus. 
I really enjoy working with you on this PR 😃

**Comment by nrichers** [comment]
@amyshwang Woot, variable interpolation works! I'll need to do a bit more testing just to be sure before pushing a commit, but adding a few variable definitions on the fly gives you what you need:

```
## Download the binaries

<InterpolateReleaseData
  renderText={(release) => {
    const anynojvmurl = `https://github.com/questdb/questdb/releases/download/${release.name}/questdb-${release.name}-no-jre-bin.tar.gz`
    const linuxurl = `https://github.com/questdb/questdb/releases/download/${release.name}/questdb-${release.name}-rt-linux-amd64.tar.gz`
    const bsdurl = `https://github.com/questdb/questdb/releases/download/${release.name}/questdb-${release.name}-rt-freebsd-amd64.tar`
    const windowsurl = `https://github.com/questdb/questdb/releases/download/${release.name}/questdb-${release.name}-rt-windows-amd64.tar.gz`
    return (
      <div>
        <ul>
          <li><b>Any (no JVM):</b>{" "}<a href={anynojvmurl} rel="noopener noreferrer" target="_blank">questdb-{release.name}-no-jre-bin.tar.gz</a></li>
          <li><b>QuestDB (Linux):</b>{" "}<a href={linuxurl} rel="noopener noreferrer" target="_blank">questdb-{release.name}-rt-linux-amd64.tar.gz</a></li>
          <li><b>QuestDB (FreeBSD):</b>{" "}<a href={bsdurl} rel="noopener noreferrer" target="_blank">questdb-{release.name}-rt-freebsd-amd64.tar.gz</a></li>
          <li><b>QuestDB (Windows):</b>{" "}<a href={windowsurl} rel="noopener noreferrer" target="_blank">questdb-{release.name}-rt-windows-amd64.tar.gz</a></li>
      </ul>
    </div>
    )
  }}
/>
```

The output looks like this: 

image

I think you pushed a commit that changes the tab labels for the different OS versions elsewhere, will take a look and match that. But basically this very simple approach seems to solve the issue and should work in future versions just fine. 

(For reference, it was [this comment](https://github.com/mdx-js/mdx/issues/1095#issuecomment-967511279) about quoting expressions that provided me with the necessary clue.)

**Comment by nrichers** [comment]
@amyshwang as promised, 4ac1afba496d5526083db440d872e5717fbcf83c adds the direct download links. Instead of the earlier version from https://github.com/questdb/questdb.io/pull/1193#issuecomment-1326759449, my commit reuses the tabs from the original for consistency with the rest of this topic: 



In testing, the downloads pull the correct tarballs for the latest 6.6.1 version:

image

If there's nothing else you need from me, can you approve and merge? And thank you for the opportunity to tinker with Docusaurus, I enjoyed our collaboration on this PR very much. 👋

**Comment by nrichers** [comment]
FYI, I added the missing FreeBSD download tile in https://github.com/questdb/questdb.io/pull/1208 — super easy fix.

**Review by amyshwang — APPROVED** [review]
Thank you for your contribution!

**Comment by nrichers** [comment]
@amyshwang thank you for approving this pull request, much appreciated. I am not able to merge ("The base branch restricts merging to authorized users"). Are you able to merge this?

**Comment by amyshwang** [comment]
Hi @nrichers,
@argshook made some suggestion to your PR by adding a PR on your forked repo - https://github.com/nrichers/questdb.io/pull/1
If you're happy with the proposed change, you could merge it then this PR will be updated. Hope this makes sense!

**Comment by nrichers** [comment]
> Hi @nrichers,
> @argshook made some suggestion to your PR by adding a PR on your forked repo - [nrichers#1](https://github.com/nrichers/questdb.io/pull/1)
> If you're happy with the proposed change, you could merge it then this PR will be updated. Hope this makes sense!

Thank you for the ping, @amyshwang! I'd missed the very nice PR from @argshook. Tested the changes locally and they look great, so I merged them into this PR branch. 

I still can't merge, so if you're happy with the current state of this PR here, can you merge?

**Review by amyshwang — APPROVED** [review]
Thank you!

**Comment by amyshwang** [comment]
PR merged - thank you again for your contribution!

**Pull request merged**